File: services/forms/app/schemas/adt1.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, EmailStr


class Adt1Base(BaseModel):
    company_id: Optional[int] = None
    cin: Optional[str] = None
    company_name: Optional[str] = None
    company_address: Optional[str] = None
    company_email: Optional[EmailStr] = None
    falling_section: Optional[str] = None
    appointment_nature: Optional[str] = None
    is_auditor_appointed: Optional[str] = None
    auditor_number: Optional[float] = None
    annual_general_meeting: Optional[str] = None
    agm_date: Optional[datetime] = None
    appointment_date: Optional[datetime] = None
    is_casual_vacancy: Optional[str] = None
    srn_of_relevant_form: Optional[str] = None
    person_vacated: Optional[str] = None
    vacancy_date: Optional[datetime] = None
    casual_vacancy_reasons: Optional[str] = None
    membership_number: Optional[float] = None
    auditor_serial: Optional[bool] = None
    auditor_member_no: Optional[str] = None
    resolution_number: Optional[str] = None
    resolution_date: Optional[datetime] = None
    signed_by: Optional[str] = None
    designation: Optional[str] = None
    din: Optional[str] = None
    ref_user_name: Optional[str] = None


class Adt1Create(Adt1Base):
    company_id: int
    cin: str
    company_name: str


class Adt1Read(Adt1Base):
    id: int
    created_at: datetime

    class Config:
        from_attributes = True

File: ComplyCrafter_Data/Forms/ADT1.cs
using AlphaUtil.Core.Models;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace ComplyCrafter_Data
{
    [TableName("tbl_adt1", "vw_adt1")]
    public class ADT1 : BaseModel
    {
        [Column("ref_user")]
        public int? RefUser { get; set; }

        [Column("company_id")]
        public int? CompanyId { get; set; }

        [Column("cin")]
        public string? Cin { get; set; }

        [Column("company_name")]
        public string? CompanyName { get; set; }

        [Column("company_address")]
        public string? CompanyAddress { get; set; }

        [Column("company_email")]
        public string? CompanyEmail { get; set; }

        [Column("falling_section")]
        public string? FallingSection { get; set; }

        [Column("appointment_nature")]
        public string? AppointmentNature { get; set; }

        [Column("is_auditor_appointed")]
        public string? IsAuditorAppointed { get; set; }

        [Column("auditor_number")]
        public decimal? AuditorNumber { get; set; }

        [Column("annual_general_meeting")]
        public string? AnnualGeneralMeeting { get; set; }

        [Column("agm_date")]
        public DateTime? AgmDate { get; set; }

        [Column("appointment_date")]
        public DateTime? AppointmentDate { get; set; }

        [Column("is_casual_vacancy")]
        public string? IsCasualVacancy { get; set; }

        [Column("sr_no_of_relevant_form")]
        public string? SrnOfRelevantForm { get; set; }

        [Column("person_vacated")]
        public string? PersonVacated { get; set; }

        [Column("vacancy_date")]
        public DateTime? VacancyDate { get; set; }

        [Column("casual_vacancy_reasons")]
        public string? CasualVacancyReasons { get; set; }

        [Column("membership_number")]
        public decimal? MembershipNumber { get; set; }

        [Column("auditor_serial")]
        public Boolean? AuditorSerial { get; set; }

        [Column("auditor_member_no")]
        public string? AuditorMemberNo { get; set; }

        [Column("resolution_number")]
        public string? ResolutionNumber { get; set; }

        [Column("resolution_date")]
        public DateTime? ResolutionDate { get; set; }

        [Column("signed_by")]
        public string? SignedBy { get; set; }

        [Column("designation")]
        public string? Designation { get; set; }

        [Column("din")]
        public string?  DIN { get; set; }
    }

    [TableName("tbl_adt1", "vw_adt1")]
    public class ADT1View : ADT1
    {
        [Column("ref_user_name")]
        public string? RefUserName { get; set; }
    }

}